cubic-server/main.cpp

```diff
@@ -1,32 +1,45 @@
 #include <iostream>
 #include <cstring>
+#include <unordered_map>
+#include <functional>
 
+#include <argparse/argparse.hpp>
 #include "Server.hpp"
 #include "protocol/ServerPackets.hpp"
 #include "interface/ManagementInterface.hpp"
 
-static void print_usage(const char *caller)
-{
-    std::cout << "Usage:\n\t" << caller << std::endl;
-}
+bool arg_parser(int argc, char **argv) {
+    argparse::ArgumentParser program("cubic_server");
+
+    program.add_argument("--nogui")
+        .help("prevents the GUI from displaying")
+        .default_value(true)
+        .implicit_value(false);
+    try {
+        program.parse_args(argc, argv);
+    }
+    catch (const std::runtime_error& err) {
+        std::cerr << err.what() << std::endl;
+        std::cerr << program;
+        std::exit(1);
+    }
 
-// Uncomment that to launch the interface
-// #define INTERFACE
+    if (program["--nogui"] == false) {
+        return false;
+    }
+    return true;
```

Comment:
For now this works because we only check the gui, but if we add more options later, it would be best to return a map of arguments, or the argument parser itself.